app/services/unlock_next_core_exercise.rb

```diff
@@ -4,7 +4,10 @@ class UnlockNextCoreExercise
   initialize_with :track, :user
 
   def call
-    return core_exercise_in_progress if core_exercise_in_progress
+    if existing_core_exercise
+      existing_core_exercise.update(mentoring_requested_at: Time.now) unless existing_core_exercise.mentoring_requested_at
```

Comment:
Yeah, [here](https://github.com/exercism/website/pull/454/files/23e6145f067f1569063b1d93fee96d6e3cc2789b#diff-da68e877c63f14f0281db6cef96f9c9dR27).

It's part of the [Mandate gem](http://github.com/ihid/mandate)

Thanks for checking :)

Reply:
Sorry I deleted my comment; I noticed `memoize` just below.

Or rather, I noticed `memoized` first, gave it no thought, asked the question, and *then* started thinking. ;-)

Reply:
Ha!